app/views/handlers/question.py

```diff
@@ -254,9 +254,8 @@ def evaluate_and_update_section_status_on_list_change(self, list_name):
         )
 
         for section_id, list_item_id in section_keys_to_evaluate:
-            path = self.router.routing_path(section_id, list_item_id)
             self.questionnaire_store_updater.update_section_status(
-                is_complete=self.router.is_path_complete(path),
+                is_complete=self.router.is_path_complete(self._routing_path),
```

Comment:
This shouldn't have changed?

Reply:
We don't want to re-evaluate the path so where is path supposed to be from if not self?

Reply:
The path of the current section in the loop. Right now this is evaluating the path for the current location for every section to evaluate.

We don't want to reevaluate the same path multiple times. Prior to your change, it would have evaluated the path for these "sections" and then when `update_progress_for_dependent_sections` was called any sections that are dependents based on any of our other dependencies would have been reevaluated. However, the sections here and the dependencies ones could overlap hence you would be evaluating the path for the same section multiple times.

Instead of evaluating the path, add it to `self._dependent_sections` which will reevaluate it when `update_progress_for_dependent_sections` is called.

Reply:
Adding this change breaks some of our existing behaviour, for example `return_to_url` parameter doesn't persist after clicking previous on edit or remove pages for list collector. So I am not sure if we should carry out this refactor? I've tried to only update dependencies instead of re-evaluating like this:
```python
        for section_id, list_item_id in section_keys_to_evaluate:
            path = self.router.routing_path(section_id, list_item_id)
            self.questionnaire_store_updater.dependent_sections.add(
                DependentSection(
                    section_id=section_id,
                    list_item_id=list_item_id,
                    is_complete=self.router.is_path_complete(path),
                )
            )
```

Reply:
What was the issue around `return_to`?

Reply:
Addressed now.